### PR TITLE
[IMP] base_user_role: UX improvements

### PR DESCRIPTION
**Pull request by kevinkhao**
Roles should not be able to be picked several times for a user. Show only relevant roles on user role lines.

Screenshot from 2021-01-26 00-34-57

Screenshot from 2021-01-26 00-35-18

### CODE REVIEW

**Comment by yostashiro** [comment]
Functional test. LGTM 👍

**Comment by kevinkhao** [comment]
ping @sebastienbeau

File: base_user_role/views/user.xml [hunk 1]
```diff
@@ -12,7 +12,11 @@
                 <page string="Roles">
                     <field name="role_line_ids" nolabel="1">
                         <tree editable="bottom" decoration-muted="not is_enabled">
-                            <field name="role_id" />
+                            <field name="user_roles_already_picked" invisible="1" />
+                            <field
+                                name="role_id"
+                                domain="[('id', 'not in', user_roles_already_picked)]"
```
Comment (sebastienbeau) [inline base_user_role/views/user.xml]: you can make it simplier.
There is already a computed field on the res.user see here https://github.com/OCA/server-backend/blob/14.0/base_user_role/models/user.py#L15

You can simply add this field in the view (in invisible) and then use the domain 
 domain="[('id', 'not in', parent.role_ids)]"

Comment (kevinkhao) [reply base_user_role/views/user.xml]: changed + functional tested

**Comment by sebastienbeau** [comment]
@kevinkhao can you fix the lint?

**Comment by kevinkhao** [comment]
@sebastienbeau done

**Review by sebastienbeau — APPROVED** [review]
LGTM code review

**Pull request merged**